src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java

```diff
@@ -162,143 +226,211 @@ public AttachCaseCallbackService(CcdApi ccdApi, PaymentsProcessor paymentsProces
+                return updateSupplementaryEvidenceWithOcr(
+                    callBackEvent,
+                    targetCaseCcdRef,
+                    exceptionRecordDetails,
+                    ignoreWarnings
+                );
+
+            default:
+                throw new CallbackException("Invalid Journey Classification: " + callBackEvent.classification);
+        }
+    }
+
+    private Either<ErrorsAndWarnings, Boolean> updateSupplementaryEvidence(
+        AttachToCaseEventData callBackEvent,
+        String targetCaseCcdRef,
+        CaseDetails exceptionRecordDetails
+    ) {
+        verifyExceptionRecordIsNotAttachedToCase(
+            callBackEvent.exceptionRecordJurisdiction,
+            callBackEvent.exceptionRecordId
+        );
 
-        verifyExceptionRecordIsNotAttachedToCase(exceptionRecordJurisdiction, exceptionRecordId);
+        CaseDetails theCase = ccdApi.getCase(targetCaseCcdRef, callBackEvent.exceptionRecordJurisdiction);
+        List<Map<String, Object>> targetCaseDocuments = getScannedDocuments(theCase);
 
         //This is done so exception record does not change state if there is a document error
         checkForDuplicatesOrElse(
-            exceptionRecordDocuments,
+            callBackEvent.exceptionRecordDocuments,
             targetCaseDocuments,
             ids -> throwDuplicateError(targetCaseCcdRef, ids)
         );
 
         List<Map<String, Object>> newCaseDocuments = attachExceptionRecordReference(
-            exceptionRecordDocuments,
-            exceptionRecordId
+            callBackEvent.exceptionRecordDocuments,
+            callBackEvent.exceptionRecordId
         );
 
-        StartEventResponse event = ccdApi.startAttachScannedDocs(theCase, idamToken, userId);
+        StartEventResponse ccdStartEvent =
+            ccdApi.startAttachScannedDocs(theCase, callBackEvent.idamToken, callBackEvent.userId);
 
         Map<String, Object> newCaseData = buildCaseData(newCaseDocuments, targetCaseDocuments);
 
         ccdApi.attachExceptionRecord(
             theCase,
-            idamToken,
-            userId,
+            callBackEvent.idamToken,
+            callBackEvent.userId,
             newCaseData,
-            createEventSummary(theCase, exceptionRecordId, newCaseDocuments),
-            event
+            createEventSummary(theCase, callBackEvent.exceptionRecordId, newCaseDocuments),
+            ccdStartEvent
         );
 
         log.info(
             "Successfully attach Exception Record with ID {} to case with id {}",
             theCase.getId(),
-            exceptionRecord.getId()
+            exceptionRecordDetails.getId()
         );
 
-        paymentsProcessor.updatePayments(exceptionRecord, theCase.getId());
+        paymentsProcessor.updatePayments(exceptionRecordDetails, theCase.getId());
 
+        return Either.right(true);
+    }
+
+    private Either<ErrorsAndWarnings, Boolean> updateSupplementaryEvidenceWithOcr(
```

Comment:
If Right, the result will always be `true`, so maybe `Optional` would be better?